src/background/tab-context-manager.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { BrowserAdapter } from '../common/browser-adapters/browser-adapter';
+import { Message } from '../common/message';
+import { BrowserMessageBroadcasterFactory } from './browser-message-broadcaster-factory';
+import { ExtensionDetailsViewController } from './extension-details-view-controller';
+import { TabToContextMap } from './tab-context';
+import { TabContextFactory } from './tab-context-factory';
+
+export class TabContextManager {
+    constructor(
+        private readonly targetPageTabIdToContextMap: TabToContextMap,
+        private readonly broadcasterFactory: BrowserMessageBroadcasterFactory,
+        private readonly browserAdapter: BrowserAdapter,
+        private readonly detailsViewController: ExtensionDetailsViewController,
+        private readonly tabContextFactory: TabContextFactory,
+        private persistStoreData = false,
```

Comment:
Can we make this non-optional? It looks like we're not passing it in anywhere at the moment, when I think we do want to pass in true from the service worker init. Either way, I think it's more clear to explicitly specify in the init files if we're persisting or not.

Reply:
Moved this flag into TabContextFactory where it's actually used and made it non-optional